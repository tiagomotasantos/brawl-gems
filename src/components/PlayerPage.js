import React, { useContext } from 'react';
import BrawlContext from '../context/brawl-context';
import BrawlStats from './BrawlStats';
import BrawlProgressBar from './BrawlProgressBar';
import BrawlClan from './BrawlClan';

const PlayerPage = () => {
    const { state } = useContext(BrawlContext);
    const { player: { data } } = state;
    const [current, max] = data.expFmt.split('/');

    return (    
        <div className="content-container">
            <div className="player-general-info">
                <div className="player-general-info__content">
                    <div className="player-general-info__image-tag">
                        <img alt={data.name} src={data.avatarUrl} className="player-general-info__image" />
                        <span className="player-general-info__tag">
                            #{data.tag}
                        </span>
                    </div>
                    <div className="player-general-info__col">
                        <span className="player-general-info__name">
                            {data.name}
                        </span>
                        <BrawlProgressBar 
                            value={data.expLevel} 
                            icon="/images/level.png" 
                            current={current} 
                            max={max} 
                            color="#3EBEED"
                        />
                        { data.hasClub && <BrawlClan icon={data.club.badgeUrl} value={data.club.name} /> }
                    </div>
                </div>
            </div>
            <div className="player-stats-container">
                <div className="player-stats-container__row">
                    <BrawlStats 
                        label="Trophies"
                        icon="/images/trophy.png" 
                        value={data.trophies}
                    />
                    <BrawlStats 
                        label="Highest Trophies"
                        icon="/images/ranking.png" 
                        value={data.highestTrophies}
                    />
                </div>
                <div className="player-stats-container__row">
                    <BrawlStats 
                        label="Solo Victories"
                        icon="/images/solo.png" 
                        value={data.soloShowdownVictories}
                    />
                    <BrawlStats 
                        label="3 vs 3 Victories"
                        icon="/images/3vs3.png" 
                        value={data.victories}
                    />
                </div>
                <div className="player-stats-container__row">
                    <BrawlStats 
                        label="Duo Victories"
                        icon="/images/duo.png" 
                        value={data.duoShowdownVictories}
                    />
                    <BrawlStats 
                        label="Best Robo Rumble Time"
                        icon="/images/robo-rumble.png" 
                        value={data.bestRoboRumbleTime}
                    />
                </div>
                <div className="player-stats-container__row">
                    <BrawlStats 
                        label="Best Time as Big Brawler"
                        icon="/images/big-brawler.png" 
                        value={data.bestTimeAsBigBrawler}
                    />
                    <BrawlStats 
                        label="Highest Power Play"
                        icon="/images/power-play.png" 
                        value={data.highestPowerPlayPoints}
                    />
                </div>
            </div>
        </div>
    );
};

export default PlayerPage;