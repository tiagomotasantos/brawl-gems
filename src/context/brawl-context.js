import React from 'react'

const BrawlContext = React.createContext()

export { BrawlContext as default }