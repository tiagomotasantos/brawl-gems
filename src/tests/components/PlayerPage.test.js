import React from 'react';
import { shallow } from 'enzyme';
import PlayerPage from '../../components/PlayerPage';
import data from '../fixtures/player';
import * as BrawlContext from '../../context/brawl-context';

let wrapper;

beforeEach(() => {
    BrawlContext.default = React.createContext({ state: { player: { data } } });
    wrapper = shallow(<PlayerPage />);
});

test('should render PlayerPage', () => {
    expect(wrapper).toMatchSnapshot();
});