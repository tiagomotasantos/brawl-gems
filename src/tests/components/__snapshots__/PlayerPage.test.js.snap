// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlayerPage 1`] = `
<div
  className="content-container"
>
  <div
    className="player-general-info"
  >
    <div
      className="player-general-info__content"
    >
      <div
        className="player-general-info__image-tag"
      >
        <img
          alt="Mottyze"
          className="player-general-info__image"
          src="https://www.starlist.pro/assets/profile/28000000.png"
        />
        <span
          className="player-general-info__tag"
        >
          #
          9RCQGYC8Y
        </span>
      </div>
      <div
        className="player-general-info__col"
      >
        <span
          className="player-general-info__name"
        >
          Mottyze
        </span>
        <BrawlProgressBar
          color="#3EBEED"
          current="402"
          icon="/images/level.png"
          max="620"
          value={59}
        />
        <BrawlClan
          icon="https://www.starlist.pro/img/club/0.png"
          value="LEGIÃƒO PT"
        />
      </div>
    </div>
  </div>
  <div
    className="player-stats-container"
  >
    <div
      className="player-stats-container__row"
    >
      <BrawlStats
        icon="/images/trophy.png"
        label="Trophies"
        value={6262}
      />
      <BrawlStats
        icon="/images/ranking.png"
        label="Highest Trophies"
        value={6277}
      />
    </div>
    <div
      className="player-stats-container__row"
    >
      <BrawlStats
        icon="/images/solo.png"
        label="Solo Victories"
        value={62}
      />
      <BrawlStats
        icon="/images/3vs3.png"
        label="3 vs 3 Victories"
        value={830}
      />
    </div>
    <div
      className="player-stats-container__row"
    >
      <BrawlStats
        icon="/images/duo.png"
        label="Duo Victories"
        value={128}
      />
      <BrawlStats
        icon="/images/robo-rumble.png"
        label="Best Robo Rumble Time"
        value="4m 24s"
      />
    </div>
    <div
      className="player-stats-container__row"
    >
      <BrawlStats
        icon="/images/big-brawler.png"
        label="Best Time as Big Brawler"
        value="2m 53s"
      />
      <BrawlStats
        icon="/images/power-play.png"
        label="Highest Power Play"
        value={316}
      />
    </div>
  </div>
</div>
`;
